src/cpp/fastdds/xtypes/utils.cpp

```diff
@@ -772,10 +755,9 @@ ReturnCode_t json_serialize_basic_member(
         }
         case TK_BITMASK:
         {
-            ReturnCode_t ret;
-
             MemberDescriptor::_ref_type bitmask_member_desc{traits<MemberDescriptor>::make_shared()};
-            if (RETCODE_OK != (ret = data->get_descriptor(bitmask_member_desc, member_id)))
+            ReturnCode_t ret = data->get_descriptor(bitmask_member_desc, member_id);
+            if (RETCODE_OK != ret)
```

Comment:
The serialization of a BITMASK is working fine when `data` refers to the BItmask itself. However, when trying to serialize a `sequence<BitMask>` it fails cause you're trying to get the descriptors of the sequence. I believe you should try to handle these to different scenarios, as in the second one the Bitmask needs to be loaned from the sequence.

Reply:
I made some changes so the bitmask case works well with both structures and collections. However after reviewing the code I found another case (collections of enumerations) with a similar issue, that I couldn't fix in the same way. I will wait for @richiware to decide whether allowing `DynamicData::get_descriptor` should also work for the collections. If not I will need to refactor the code a bit.

Reply:
Eventually we declined extending `DynamicData::get_descriptor` to work with collections, as it was incompatible with XTypes standard. Instead, the type descriptor is obtained differently depending on the "container" kind.